Sources/Shared/Core/SessionManaging.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  VimeoSessionManaging.swift
+//  VimeoNetworking
+//
+//  Created by Rogerio de Paula Assis on 9/5/19.
+//  Copyright © 2019 Vimeo. All rights reserved.
+//
+
+import Foundation
+
+/// The protocols declared in this file have been created to abstract our dependency
+/// on AFNetworking and the Vimeo subclasses that inherit from it,
+/// Specifically `VimeoSessionManager`, `VimeoRequestSerializer` and `VimeoResponseSerializer`
+/// The goal is to make it easier for these dependencies to be swapped out when needed.
+public protocol AuthenticationListeningDelegate {
```

Comment:
Any particular reason these are separate? Looks like they are only used as part of a compound existential. I can see that these methods are conceptually a bit different (and depend on VimeoNetworking), so totally down for this separation. Just curious if there was any stronger reason.

Reply:
Relatedly—does this protocol need to be public? Looks like the methods in `VimeoSessionManager` that this abstracts over were internal previously. Why does that need to change?

Reply:
The reason is purely conceptual, even the naming was difficult when I tried to lump them all together.

The reason they had to be made public is because of the public initializer that exposes the compound predicate.

Reply:
Makes sense to me. Merge when ready!